pkg/startup/startup_linux.go

```diff
@@ -100,3 +103,41 @@ func ipv6Supported() bool {
 func ensureNetworkForOS(ctx context.Context, client client.Interface, nodeName string) error {
 	return nil
 }
+
+func MonitorIPAddressSubnets() {
+	autoDetectPollingInterval := DEFAULT_AUTODETECT_POLL_INTERVAL
+	if os.Getenv("AUTODETECT_POLL_INTERVAL") != "" {
+		autoDetectPollingInterval, _ = time.ParseDuration(os.Getenv("AUTODETECT_POLL_INTERVAL"))
+	}
+
+	// Add a subscription to get updated if there is any change to interface addresses.
+	addrUpdate := make(chan netlink.AddrUpdate)
+	done := make(chan struct{})
+	if err := netlink.AddrSubscribe(addrUpdate, done); err != nil {
+		log.WithError(err).Error("Failed to subscribe to network interface updates")
```

Comment:
If we hit this error do we want to exit?

Reply:
I wasn't sure if exit() would resolve issue with AddrSubscribe() failure, so proposed to log the error and proceed with time-out logic. If the error is unrecoverable, startup will get stuck into a restart loop. Let me know what you think.

Reply:
Yeah, good point. I'm not sure what the "best" behavior is here.